pkg/cmd/diagnostics/diagnostics_list.go

```diff
@@ -0,0 +1,60 @@
+package diagnostics
+
+import (
+	"encoding/json"
+
+	"github.com/spf13/cobra"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-cli/pkg/cli"
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+	"github.com/dustin/go-humanize"
+
+	diagnostics "github.com/dcos/dcos-core-cli/pkg/diagnostics/v2"
+	"github.com/dcos/dcos-core-cli/pkg/pluginutil"
+)
+
+func listDiagnosticsBundles(ctx api.Context, c *httpclient.Client, jsonOutput bool) error {
+	client := diagnostics.NewClient(c)
+	bundles, err := client.List()
+	if err != nil {
+		return err
+	}
+
+	if jsonOutput {
+		enc := json.NewEncoder(ctx.Out())
+		enc.SetIndent("", "    ")
+		return enc.Encode(bundles)
+	}
+
+	tableHeader := []string{"ID", "STATUS", "STOPPED", "SIZE"}
+	table := cli.NewTable(ctx.Out(), tableHeader)
+
+	for _, b := range bundles {
+		size := humanize.Bytes(uint64(b.Size))
+		stopped := humanize.Time(b.Stopped)
+
+		fields := []string{b.ID, b.Status.String(), stopped, size}
+		table.Append(fields)
+	}
+
+	table.Render()
+
+	return nil
+}
+
+func newDiagnosticsListCommand(ctx api.Context) *cobra.Command {
+	var jsonOutput bool
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "Lists available bundles",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := pluginutil.HTTPClient("")
+			return listDiagnosticsBundles(ctx, c, jsonOutput)
```

Comment:
As this is only called once, we can have the code of `listDiagnosticsBundles` in `RunE` directly.